cudamapper/src/overlapper.cpp

```diff
@@ -38,8 +45,118 @@ void Overlapper::update_read_names(std::vector<Overlap>& overlaps,
     }
 }
 
-void Overlapper::print_paf(const std::vector<Overlap>& overlaps)
+void Overlapper::align_overlaps(std::vector<Overlap>& overlaps,
+                                const claragenomics::io::FastaParser& query_parser,
+                                const claragenomics::io::FastaParser& target_parser,
+                                int32_t num_batches,
+                                std::vector<std::string>& cigar)
 {
+    // Calculate max target/query size in overlaps
+    int32_t max_query_size  = 0;
+    int32_t max_target_size = 0;
+    for (const auto& overlap : overlaps)
+    {
+        int32_t query_overlap_size  = overlap.query_end_position_in_read_ - overlap.query_start_position_in_read_;
+        int32_t target_overlap_size = overlap.target_end_position_in_read_ - overlap.target_start_position_in_read_;
+        if (query_overlap_size > max_query_size)
+            max_query_size = query_overlap_size;
+        if (target_overlap_size > max_target_size)
+            max_target_size = target_overlap_size;
+    }
+
+    // Heuristically calculate max alignments possible with available memory based on
+    // empirical measurements of memory needed for alignment per base.
+    const float memory_per_base = 0.03f; // Estimation of space per base in bytes for alignment
+    float memory_per_alignment  = memory_per_base * max_query_size * max_target_size;
```

Comment:
Any reason you use `float` here, but integers for the rest of the calculation?

Reply:
at some point had to switch to integers since batch size and everything is integers. and just chose switch to integers at the max_alignment calculation stage and keep calculation in float till then

Reply:
I actually meant: Why do you use float? :D

Reply:
for storing the `memory_per_base` value mostly which is sub 1

Reply:
ok, since it is a heuristic calculation, it makes sense.